### PR TITLE
feat: add vault common components

### PR DESCRIPTION
**Pull request by soulbird**
### Description

As part of https://github.com/apache/apisix/issues/8319:

Add vault common component. Currently only supporting the get operation is enough

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: ci/init-common-test-service.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# prepare vault kv engine
+sleep 3s
```
Comment (moonming) [inline ci/init-common-test-service.sh:20]: why sleep 3s here?

Comment (soulbird) [reply ci/init-common-test-service.sh:20]: Make sure that the vault service has been started normally, this is just an experience value, it does work.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -33,7 +33,7 @@ jobs:
           - t/plugin/[a-k]*
           - t/plugin/[l-z]*
           - t/admin t/cli t/config-center-yaml t/control t/core t/debug t/deployment t/discovery t/error_page t/misc
-          - t/node t/pubsub t/router t/script t/stream-node t/utils t/wasm t/xds-library t/xrpc
+          - t/node t/pubsub t/router t/script t/stream-node t/utils t/wasm t/xds-library t/xrpc t/kms
```
Comment (spacewander) [inline .github/workflows/build.yml:36]: Why add the dir `k` at the end?

File: apisix/kms/vault.lua [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+--- Vault Tools.
+--  Vault is an identity-based secrets and encryption management system.
+
+local core = require("apisix.core")
+local http = require("resty.http")
+local json = require("cjson")
```
Comment (spacewander) [inline apisix/kms/vault.lua:23]: Why don't we use core.json?

File: apisix/kms/vault.lua [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    end
+
+    return res.body
+end
+
+-- key is the vault kv engine path
+local function get(conf, key)
+    core.log.info("fetching data from vault for key: ", key)
+
+    local idx = find(reverse(key), "/")
```
Comment (spacewander) [inline apisix/kms/vault.lua:62]: We can use the rfind_char instead?
https://github.com/apache/apisix/blob/164a3222fa8ef159a21726fb4a6bd7ab1ccb785c/apisix/core/string.lua#L86

**Review by spacewander — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**